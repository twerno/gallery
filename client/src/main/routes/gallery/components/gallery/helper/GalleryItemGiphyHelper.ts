import { IGiphyImage } from "@shared/";
import { IResponsivePictureProps } from "../GalleryResponsivePicture";

export default {
    convertToResponsivePictureProps
};

function convertToResponsivePictureProps(image: IGiphyImage): IResponsivePictureProps {
    const fixed_still = image.images.fixed_height_still;
    const small_still = image.images.fixed_height_small_still;

    return {
        alt: image.title,
        defaultImgSrc: small_still?.url,
        responsiveSrcList: [
            { src: fixed_still?.url, media: '(min-width: 600px)' },
        ]
    };
}
