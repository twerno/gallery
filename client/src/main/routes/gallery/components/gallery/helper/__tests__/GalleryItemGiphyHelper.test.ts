import '@testing-library/jest-dom/extend-expect';
import { IGiphyImage } from '@shared/';
import GalleryItemGiphyHelper from '../GalleryItemGiphyHelper';

describe('GalleryItemGiphyHelper', () => {

    test('convertToResponsivePictureProps', async () => {
        const source: IGiphyImage = {
            embed_url: 'embed_url',
            id: 'id',
            images: {
                fixed_height_small_still: {
                    height: '200',
                    url: 'http://example.com.not/image-150.jpg',
                    width: '150'
                },
                original: {
                    height: '300',
                    url: 'http://example.com.not/image-200.jpg',
                    width: '200'
                },
                fixed_height_still: {
                    height: '200',
                    url: 'http://example.com.not/image-300.jpg',
                    width: '300'
                },
            },
            title: 'title',
            type: 'type'
        };

        const responsivePicture = GalleryItemGiphyHelper.convertToResponsivePictureProps(source);
        expect(responsivePicture).toMatchSnapshot();
    });
});

