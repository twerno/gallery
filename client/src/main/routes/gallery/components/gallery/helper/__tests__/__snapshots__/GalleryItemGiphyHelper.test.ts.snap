// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItemGiphyHelper convertToResponsivePictureProps 1`] = `
Object {
  "alt": "title",
  "defaultImgSrc": "http://example.com.not/image-150.jpg",
  "responsiveSrcList": Array [
    Object {
      "media": "(min-width: 600px)",
      "src": "http://example.com.not/image-300.jpg",
    },
  ],
}
`;
